### PR TITLE
fix: [CDS-49973]: Added support for allowed values in Artifact and Artifact Sources 

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx [hunk 1]
```diff
@@ -312,7 +313,9 @@ function FormComponent(
               }}
             />
             {getMultiTypeFromValue(formik.values.project) === MultiTypeInputType.RUNTIME && (
-              <ConfigureOptions
+              <SelectConfigureOptions
+                options={getItems(fetchingProjects, 'Projects', projectItems)}
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx:317]: add `Projects` to strings as it's a label

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx [hunk 2]
```diff
@@ -381,7 +384,9 @@ function FormComponent(
             }}
           />
           {getMultiTypeFromValue(formik.values.feed) === MultiTypeInputType.RUNTIME && (
-            <ConfigureOptions
+            <SelectConfigureOptions
+              options={getItems(fetchingFeeds, 'Feeds', feedItems)}
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx:388]: add to strings

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx [hunk 3]
```diff
@@ -464,7 +469,9 @@ function FormComponent(
             }}
           />
           {getMultiTypeFromValue(formik.values.package) === MultiTypeInputType.RUNTIME && (
-            <ConfigureOptions
+            <SelectConfigureOptions
+              options={getItems(fetchingPackages, 'Packages', packageItems)}
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx:473]: add to strings

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx [hunk 4]
```diff
@@ -533,7 +540,9 @@ function FormComponent(
               }}
             />
             {getMultiTypeFromValue(formik.values.version) === MultiTypeInputType.RUNTIME && (
-              <ConfigureOptions
+              <SelectConfigureOptions
+                options={getItems(fetchingVersions, 'Versions', versionItems)}
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx:544]: add to strings

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx [hunk 5]
```diff
@@ -428,12 +429,14 @@ const Content = (props: ACRRenderContent): JSX.Element => {
...
-            <ExperimentalInput
+            <SelectInputSetView
+              fieldPath={`artifacts.${artifactPath}.spec.subscriptionId`}
+              template={template}
               formik={formik}
               disabled={loadingSubscriptions || isFieldDisabled(`artifacts.${artifactPath}.spec.subscriptionId`)}
               multiTypeInputProps={{
                 onChange: /* istanbul ignore next */ (
-                  value: SelectOption,
+                  value: any,
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:439]: Can the correct type be specified here?

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx [hunk 6]
```diff
@@ -487,12 +490,14 @@ const Content = (props: ACRRenderContent): JSX.Element => {
...
-            <ExperimentalInput
+            <SelectInputSetView
+              fieldPath={`artifacts.${artifactPath}.spec.registry`}
+              template={template}
               formik={formik}
               disabled={loadingRegistries || isFieldDisabled(`artifacts.${artifactPath}.spec.registry`)}
               multiTypeInputProps={{
                 onChange: /* istanbul ignore next */ (
-                  value: SelectOption,
+                  value: any,
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:500]: Can the correct type be specified here?

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx (hunk 5 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:439]: @rameshx this need `AcceptableValue` to be exported from uicore. Will do this in another PR.

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx (hunk 6 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:500]: @rameshx this need AcceptableValue to be exported from uicore. Will do this in another PR.

Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:500]: We can use ExpressionAndRuntimeTypeProps['value']

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx (hunk 6 above)
Comment (pankajsingh-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ACRArtifactSource/ACRArtifactSource.tsx:500]: Done.

**Review by rameshx — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx (hunk 1 above)
Comment (pankajsingh-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx]: Done

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx (hunk 2 above)
Comment (pankajsingh-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx]: Done

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx (hunk 3 above)
Comment (pankajsingh-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx]: Done

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx (hunk 4 above)
Comment (pankajsingh-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AzureArtifacts/AzureArtifacts.tsx]: Done

**Review by rameshx — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**